solr/solrj/src/test/org/apache/solr/client/solrj/impl/BasicHttpSolrClientTest.java

```diff
@@ -476,9 +484,20 @@ public void testUpdate() throws Exception {
       // parameter encoding
       assertEquals(1, DebugServlet.parameters.get("a").length);
       assertEquals("\u1234", DebugServlet.parameters.get("a")[0]);
+    }
+    DebugServlet.clear();
+    // XML response and writer
+    try (HttpSolrClient client =
+        new HttpSolrClient.Builder(url)
+            .withRequestWriter(new RequestWriter())
+            .withResponseParser(new XMLResponseParser())
+            .build()) {
+      UpdateRequest req = new UpdateRequest();
+      req.add(new SolrInputDocument());
+      req.setParam("a", "\u1234");
 
-      // XML response and writer
-      client.setParser(new XMLResponseParser());
+      // Have not been able to move this to the Builder pattern, the below check for application/xml
+      // always returns as application/javabin when .setRequestWriter() is commented out.
```

Comment:
thanks for calling this out.   With fresh eyes, I realized, the test was failing PROPERLY, as the code wasn't right...   So I fixed it finally.  Phew, it would have been an annoying bug.